src/Microsoft.AspNetCore.Authentication.Core/AuthenticationSchemeProvider.cs

```diff
@@ -167,6 +168,14 @@ public void AddScheme(AuthenticationScheme scheme)
                 {
                     _requestHandlers.Add(scheme);
                 }
+                if (typeof(IAuthenticationSignInHandler).IsAssignableFrom(scheme.HandlerType))
+                {
+                    _signInHandlers.Add(scheme);
+                }
+                if (typeof(IAuthenticationSignOutHandler).IsAssignableFrom(scheme.HandlerType))
+                {
+                    _signOutHandlers.Add(scheme);
+                }
                 _map[scheme.Name] = scheme;
```

Comment:
That's done in the scheme itself, https://github.com/aspnet/HttpAbstractions/blob/88a797cffddb43e586def38acc17c9435367a19a/src/Microsoft.AspNetCore.Authentication.Abstractions/AuthenticationScheme.cs#L31